### PR TITLE
[vcpkg] update gas-preprocessor and clang in vcpkg_find_acquire_program

### PR DESCRIPTION
**Pull request by reitowo**
**Describe the pull request**

Update `gas-preprocessor` and `clang` in `vcpkg_find_acquire_program`
Because if this update is included in port update PR, the CI will rebuild all ports and takes forever, so I fan out this PR.

### CODE REVIEW

File: scripts/cmake/vcpkg_find_acquire_program.cmake [hunk 1]
```diff
@@ -425,14 +425,14 @@ function(vcpkg_find_acquire_program program)
     elseif(program STREQUAL "GASPREPROCESSOR")
         set(raw_executable true)
         set(program_name gas-preprocessor)
-        set(tool_subdirectory "4daa6115")
+        set(tool_subdirectory "9309c67a")
```
Comment (Neumann-A) [inline scripts/cmake/vcpkg_find_acquire_program.cmake:428]: probably should use `string(SUBSTRING)` and get that from `download_sha512`

Comment (cnSchwarzer) [reply scripts/cmake/vcpkg_find_acquire_program.cmake]: How about:
```cmake
set(commit_id 9309c67acb535ca6248f092e96131d8eb07eefc1)
set(download_urls "https://raw.githubusercontent.com/FFmpeg/gas-preprocessor/${commit_id}/gas-preprocessor.pl")
string(SUBSTRING ${commit_id} 0 8 tool_subdirectory)
```

**Comment by cnSchwarzer** [comment]
The regression of `x64_windows_static` seems have nothing to do with this PR as that port doesn't even use `vcpkg_find_acquire_program` at all.

**Comment by cnSchwarzer** [comment]
The regression of `x64_osx` seems have nothing to do with this PR as that port doesn't even use vcpkg_find_acquire_program at all. Can we merge this PR? @FrankXie05

**Comment by FrankXie05** [comment]
@cnSchwarzer I'm afraid we can only merge this PR if the ci is resolved. 😣

**Comment by cnSchwarzer** [comment]
OK, I'll wait until these ports are resolved

**Comment by FrankXie05** [comment]
We will solve the problem of ci as soon as possible and we have to re-run ci to ensure that the problem is solved.

**Comment by cnSchwarzer** [comment]
> We will solve the problem of ci as soon as possible

Do you mean only rebuild ports that have reference to specific changed program? For example only rebuild ports that acquires program CLANG and GAS_PREPROCESSOR in this case.

**Comment by dg0yt** [comment]
> For example only rebuild ports that acquires program CLANG and GAS_PREPROCESSOR in this case. Thats seems difficult

It is more or less what vcpkg CI already does. It determines the ports which are directly *and transitively* affected by your changes ("cone of destruction"), and rebuilds them if not cached. For this purpose, it must also install all unaffected ports which are dependencies. The total set of ports can pull in known baseline regressions, or uncover regressions for particular feature configurations or installation orders.

**Comment by cnSchwarzer** [comment]
@FrankXie05 Can you rerun the ci? I think the problem is fixed

**Comment by cnSchwarzer** [comment]
REGRESSION: xxhash:* failed with FILE_CONFLICTS

**Comment by cnSchwarzer** [comment]
OK checked passed, let's merge this please

**Comment by cnSchwarzer** [comment]
@FrankXie05

**Review by FrankXie05 — APPROVED** [review]

**Pull request merged**